### PR TITLE
Add is-global with global utilities

### PR DESCRIPTION
**Pull request by isaacroldan**
### WHY are these changes introduced?
As part of the global cli we need some utilities to detect and work with the global CLI.

### WHAT is this pull request doing?
Add some utilities to cli-kit to detect and install CLI, or infer the global CLI package manager.

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/cli-kit/src/public/node/is-global.ts [hunk 1]
```diff
@@ -9,3 +14,70 @@ export function currentProcessIsGlobal(env = process.env): boolean {
...
 }
+
+/**
+ * Returns true if the global CLI is installed.
+ *
+ * @returns `true` if the global CLI is installed.
+ */
+export async function isGlobalCLIInstalled(): Promise<boolean> {
+  try {
+    await captureOutput('shopify', ['--help'])
```
[the remark this answers is not in the document]
Comment (gonzaloriestra) [reply packages/cli-kit/src/public/node/is-global.ts:25]: What about this? https://github.com/Shopify/cli/blob/9e3c217b407c19c5bb44abf86802ce3819cc535d/bin/create-test-theme.js#L81

[the remark this answers is not in the document]
Comment (isaacroldan) [reply packages/cli-kit/src/public/node/is-global.ts]: We can't check for that, we are not installing `@shopify/app` anymore, all the commands are bundled inside `@shopify/cli`

My checking for the presence of `app dev` in the string returned after running `shopify app`. That's something only available in Global CLI

[the remark this answers is not in the document]
Comment (gonzaloriestra) [reply packages/cli-kit/src/public/node/is-global.ts]: Ok, makes sense then! One more alternative (just looking for something faster): `npm ls -g | grep @shopify/cli` and check that the version is greater than X?

[the remark this answers is not in the document]
Comment (isaacroldan) [reply packages/cli-kit/src/public/node/is-global.ts]: We could do that, but just tested it and is actually slower than running `shopify app`

[the remark this answers is not in the document]
Comment (gonzaloriestra) [reply packages/cli-kit/src/public/node/is-global.ts]: Perfect. That's because the CLI is blazing fast now! 🚀

**Pull request merged**